components/tableIndexer/table_reader.py

```diff
@@ -0,0 +1,118 @@
+import os
+from utils.table_processing import get_cell_coordinates_by_row, apply_ocr
+import easyocr
+import sys
+import json
+import logging
+import argparse
+from utils.logger import setup_logging
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="read the content of the table by passing the json file which contains the cells details"
+    )
+    parser.add_argument(
+        "--img-path",
+        dest="img_path",
+        required=True,
+        help="Specify the path to the input image",
+    )
+    parser.add_argument(
+        "--json-path",
+        dest="json_path",
+        required=True,
+        help="Specify the path of the json file containing the cells result",
+    )
+    parser.add_argument(
+        "--languages",
+        dest="languages",
+        nargs="+",
+        default=["en"],
+        help="an array of langauges that has to be supported to read the document. Example: ['en']",
+    )
+    parser.add_argument(
+        "--json",
+        dest="save the output as a json file",
+        default=False,
+        help="set the json output preference",
+    )
+    parser.add_argument(
+        "--output-path",
+        dest="specify the json path",
+        default="./outputs/parsed_tables_json",
+        help="set the output path of the json file",
+    )
+    parser.add_argument(
+        "--log-level",
+        dest="log_level",
+        default="INFO",
+        help="Set the logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)",
+    )
+
+    return parser.parse_args()
+
+
+def table_reader(
+    img_path, json_path, languages=["en"], output_path="./outputs/parsed_tables"
+):
+    try:
+        logging.info("✨ Sentinel Table Reader has started...")
+
+        reader = easyocr.Reader(languages)
+
+        logging.debug("✔️  EasyOCR Model loaded.")
+        logging.debug(f"🟡 Model Language: {languages}")
+
+        # load json file
+        if os.path.exists(json_path):
```

Comment:
**Suggestion: catch the `FileNotFoundError` exception**

instead of an explicit check, more pythonic imho